constellations/gemini-admin/src/achieve/achieveEdit.js

```diff
@@ -143,35 +115,66 @@ export class AchieveEditPage extends React.Component {
                 />
             );
         }
+        return modalDiv;
+    };
+
+    render = () => {
+        const isEdit = this.state.isEdit;
+        const title = isEdit ? "Edit Achievement" : "Add Achievement";
+
+        let deleteButton = isEdit ? (
+            <button className="btn-delete" onClick={this.onClickDelete}>
+                Delete
+            </button>
+        ) : (
+            <div></div>
+        );
 
         return (
             <div id="view-achieve-edit">
-                {modalDiv}
+                {this.renderModal()}
                 <h2>{title}</h2>
+
                 <InputText
                     label="Year"
                     description="Enter the achievement year"
                     required={true}
-                    value={this.state.inputYear}
-                    onChangeCallback={(e) => this.handleChange(e, "inputYear")}
+                    value={this.state.year}
+                    onChangeCallback={(e) => this.handleChange(e, "year")}
                     validators={[
                         {
-                            validate: (text) => parseInt(text) > 2000,
+                            validate: (number) => parseInt(number) > 2000,
```

Comment:
shouldn't need parseInt if year is guaranteed to always be a number.

Reply:
When you input a value, it is always a string at first.

Reply:
oh, that's right. Sorry my bad.